CMakeLists.txt

```diff
@@ -67,6 +66,7 @@ add_library(${PROJECT_NAME}
   src/visualization.cpp
 )
 add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS} ${PROJECT_NAME}_gencfg)
+target_compile_options(${PROJECT_NAME} PUBLIC -std=c++17 -Wall -Wextra -Wno-unused-parameter -Werror)
```

Comment:
Is there a benefit to do this on a per-target basis? Instead of the complete scope of the package like it was before?

Reply:
As far as I know, this is a CMake best practice. Other libs/packages might need different compiler settings. Especially now we're making the compiler stricter. Other nobleo libs/packages might not build with that.

Reply:
But other libs/packages are not influenced by the `add_compile_options` in this CMakeLists.txt. It's confined to this package only.

We do have packages with lots of nodes, I think having the same settings within a package makes the most sense. Otherwise it becomes a tedious effort to align the settings even withing a package.

Reason for asking is that `add_compile_options` is the recommended ROS practice (I'll try and find the source for this later).

Reply:
Fixed.